frontend/src/Legend.js

```diff
@@ -0,0 +1,83 @@
+import React, { useState } from "react";
+import { ReactComponent as StudyIcon } from "./study-icon.svg";
+import { ReactComponent as EatingIcon} from "./eating-icon.svg";
+import { ReactComponent as RelaxingIcon} from "./relaxing-icon.svg";
+import {ReactComponent as ExpandIcon} from "./expand-icon.svg"
+
+function Legend(props) {
+
+    const [study, setStudy] = useState(false);
+    const [eating, setEating] = useState(false);
+    const [relaxing, setRelaxing] = useState(false);
+    const [expand, setExpand] = useState(true);
+
+    function handleChange(event) {
+        const { name } = event.target;
+        if (name === "study") setStudy(!study);
+        else if (name === "eating") setEating(!eating);
+        else if (name === "relaxing") setRelaxing(!relaxing);
+        else if (name === "expand"){
+            setExpand(!expand);
+            toggleExpand();
+        }
+        
+      }
+
+    function toggleExpand(){
+        var legend = document.getElementById("legend-attributes");
+        // var legend = document.getElementById("legend");
+        if (expand) {
+            legend.hidden = true;
+        }
+        else {
+            legend.hidden = false;
+        }
+    }
+
+  return (
+      <div>
+        <button type="button" class="button" id="expand" name="expand" onClick={handleChange}>
+            <ExpandIcon onClick={handleChange}/>
+        </button>
+        <div id="legend-attributes">
```

Comment:
I think you could condense code and avoid directly calling the DOM API by supplying the expand state variable to the div's hidden attribute in line 42:

<div id="legend-attributes" hidden={expand} >

Reply:
Got it, I'll make the change.